app.rb

```diff
@@ -0,0 +1,122 @@
+require_relative './book'
+require_relative './student'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  attr_reader :state
+
+  def initialize
+    @state = { students: [], teachers: [], books: [] }
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def run
+    puts 'Welcome to School Library App!'
+    running = true
+    while running
+      app_ui
+      user_selection = gets.chomp.to_i
+      case user_selection
+      when 1
+        puts list_books
+      when 2
+        puts list_people
+      when 3
+        add_person
+      when 4
+        add_book(collect_data(:book))
+      when 5
+        puts adding_rental_helper
+      when 6
+        print 'ID of the person: '
+        puts list_person_rentals(gets.chomp.to_i)
+      when 7
+        running = false
+      else
+        puts 'Invalid selection, try again!'
+      end
+    end
+  end
+  # rubocop:enable Metrics/MethodLength
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  private
+
+  def list_people
+    people = @state[:students] + @state[:teachers]
+    people.map { |person| "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age} " }
+  end
+
+  def list_books
+    @state[:books].map { |book| "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def add_book(title:, author:)
+    @state[:books] << Book.new(title: title, author: author)
+  end
```

Comment:
- [ ] When a person has been successfully created, there should be a prompt to let the user know the person has been created successfully. Something like `"Book created successfully"`